File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/TextAboveCreatureEffect.ts
﻿import { FairyGUI } from "csharp";
import Color from "../DataDefine/Color";
import { View_TextAboveEffect } from "../Gen/View/ModTheSpire_Effect";
import { AbstractEffect } from "./AbstractEffect";


export class TextAboveCreatureEffect extends AbstractEffect{
    private Interval = 2.2;
    private TargetOffestY = 80;
    public TextComponent:FairyGUI.GComponent;
    public constructor(private x:number,private y:number,private msg:string,private targetColor:Color) {
        super();
        this.TextComponent = <FairyGUI.GComponent>View_TextAboveEffect.CreateInstance();
        this.TextComponent.SetPosition(x,y,0);
        this.TextComponent.text = msg;
        let transition = this.TextComponent.GetTransition("t0");
        transition.SetValue("FromColor",targetColor.UnityColor);
        transition.Play(1,0,()=>{
            this.IsDone = true;
        });
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Other/TextAboveCreatureEffect.cs
using ModTheSpire.ModTheSpire_Effect;
using UnityEngine;

namespace ModTheSpire.Effect.Other
{
    public class TextAboveCreatureEffect : AbstractEffect
    {
        private float Interval = 2.2f;
        private int TargetOffestY = 80;
        public FairyGUI.GComponent TextComponent;

        public TextAboveCreatureEffect(float x, float y, string msg, Color targetColor)
        {
            TextComponent = View_TextAboveEffect.CreateInstance();
            TextComponent.SetPosition(x, y, 0);
            TextComponent.text = msg;
            var transition = this.TextComponent.GetTransition("t0");
            transition.SetValue("FromColor", targetColor);
            transition.Play(1, 0, Finish);
        }

        public override void Dispose()
        {
            TextComponent.Dispose();
        }
    }
}